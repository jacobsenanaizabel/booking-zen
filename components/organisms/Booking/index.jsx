import React, { useState, useEffect } from "react";
import { useSnackbar } from 'react-simple-snackbar';
import FullCalendar from "@fullcalendar/react";
import dayGridPlugin from "@fullcalendar/daygrid";
import timeGridPlugin from "@fullcalendar/timegrid";
import interactionPlugin from "@fullcalendar/interaction";
import dataEvents from "./mockedCalendar.json";


import {
	ContainerCalendar
  } from './styles';

const unavailable = "rgb(177, 160, 238)";

export const options = {
	position: 'top-center',
	style: {
	  backgroundColor: '#4a31a1',
	  color: '#ffffff',
	  fontSize: '20px',
	  textAlign: 'center',
	},
	closeStyle: {
	  color: '#ffffff',
	  fontSize: '16px',
	},
  };

const Booking = () => {
	let [convertedEvents, setConverted] = useState();
	const [openSnackbar] = useSnackbar(options)

	function handleDateClick() {
	}

	function handleEventClick(info) {
		toggleAvailabilityByProps(info);
	}

	function toggleAvailabilityByProps(info) {
		if (info.event.extendedProps.available) {
			info.el.style.backgroundColor = unavailable;
			info.el.style.borderColor = unavailable;
			const day = String(info.event.start.getDate()).padStart(2, "0");
			const month = String(info.event.start.getMonth()).padStart(2, "0");
			const year = String(info.event.start.getFullYear()).padStart(2, "0");
			const hour = String(info.event.start.getHours()).padStart(2, "0");
			const minutes = String(info.event.start.getMinutes()).padStart(2, "0");
			info.event.setExtendedProp("available", false);

			openSnackbar(`scheduled: ${day}/${month}/${year} ${hour}:${minutes}hrs`)
		} else {
			openSnackbar("Unavailable time");
		}
	}

	useEffect(() => {
		dataEvents.map((c) => {
			if (!c.available) {
				c.color = unavailable;
			}
		});

		setConverted(dataEvents);
	}, [dataEvents]);

	return (
		<>
			{convertedEvents && (
				<ContainerCalendar>
				<FullCalendar
					plugins={[dayGridPlugin, timeGridPlugin, interactionPlugin]}
					initialView="timeGridWeek"
					hiddenDays={[0, 6]}
					allDaySlot={false}
					height= "22rem"
					width="31rem"
					aspectRatio="3"
					expandRows={true}
					header={{
						center: 'title',
						right: 'prev,next'
					}}
					headerToolbar={{
						start: 'title', 
						end: 'prev,next', // will normally be on the right. if RTL, will be on the left
					  }}
					slotMinTime="08:00:00"
					slotMaxTime="18:00:00"
					events={convertedEvents}
					eventColor="#7655e9"
					dateClick={handleDateClick}
					eventClick={handleEventClick}
				/>
				</ContainerCalendar>
			)}
		</>
	);
};
export default Booking