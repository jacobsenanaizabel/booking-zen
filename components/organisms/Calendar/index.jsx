import React from 'react'
import FullCalendar from '@fullcalendar/react'
import timeGridPlugin from "@fullcalendar/timegrid";
import interactionPlugin from "@fullcalendar/interaction"

export default () => (
	<FullCalendar
	  plugins={[interactionPlugin, timeGridPlugin]}
	  initialView='timeGridWeek'
	  nowIndicator={true}
	  editable={true}
	  initialEvents={[
		{ title: 'nice event', start: new Date() }
	  ]}
	/>
  )