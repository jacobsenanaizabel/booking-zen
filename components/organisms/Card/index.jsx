import React from 'react';
import PropTypes from 'prop-types';
import { Avatar, Price, Description, Info } from '@components/atoms';
import  Rating  from '@components/atoms/Rating';

import {
  Container,
  About,
  Profile,
  TextStyled
} from './styles';

const Card = ({name, img, location, profession, price, rating, ratingNumbers, description}) => (
  <>
  <Container>
  <Profile>
    <Avatar img={img}/>
    <About>
        <TextStyled text={name} size="1.5rem" color="black"/>
        <Info location={location} profession={profession} />
        <Rating rating={rating} ratingNumbers={ratingNumbers}></Rating>
        <Price mount={price}/>
    </About>
  </Profile>
    <Description text={description}/>
  </Container>
  </>
);

Card.propTypes = {
  name: PropTypes.string,
  location: PropTypes.string,
  profession: PropTypes.string,
  price: PropTypes.number,
  rating: PropTypes.number,
  ratingNumbers: PropTypes.string,
  img: PropTypes.string
};

Card.defaultProps = {
  name: '',
  location: '',
  profession: '',
  price: 0,
  rating: 1,
  ratingNumbers: '0',
  img: '/'
};

export default Card;
