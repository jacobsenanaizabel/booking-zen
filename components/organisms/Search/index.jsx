import React, {useState } from 'react';
import { Button, Input } from '@components/atoms';

import {
  Container
} from './styles';

const Search = ({searchFreelancer}) => {

  const [search, setSearch] = useState();

  function onChangeSearching(evt){
    setSearch(evt.target.value)
  }

  const getFreelancers = (() =>{
    if(search !== undefined && search.length > 0){
      searchFreelancer(search)
    }
  })

  function onKeyDown(evt){
    if (evt.key === 'Enter') {
      searchFreelancer(evt.target.value)
    }
  }


  return(
    <>
        <Container>
          <Input
              name="search"
              onChange={onChangeSearching}
              onKeyDown={onKeyDown}
              placeholder="Find an expert by specialty or name" 
              type="text"
              value={search}
            />
          <Button text="Search" onClick={getFreelancers}/>
        </Container>
    </>
  )
};


export default Search;
