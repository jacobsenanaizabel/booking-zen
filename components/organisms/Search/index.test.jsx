import React from 'react';
import { render, fireEvent, screen  } from "@testing-library/react";
import {Search} from '@components/organisms/index';

describe('Search tests', () => {
  it('should match snapshot with default Search', () => {
    
    const { asFragment } = render(<Search />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot with default Search with searchFreelance', () => {

    const props = {
        img: 'http:img.com',
        name: 'test',
        location: 'now',
        rating: 5,
        ratingNumbers: '2 comments',
        price: 100,
        profession: 'coach'
    }
    const { asFragment } = render(<Search {...props}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should change value in input', () => {

    const setSearch = jest.fn(() => {})

    const { queryByPlaceholderText } = render(<Search onChange={setSearch}/>)

    const searchInput = queryByPlaceholderText('Find an expert by specialty or name')

    fireEvent.change(searchInput, { target: { value: 'test' } })

    expect(searchInput.value).toBe('test')
    
  });

  it('should call search freelancers', () => {

    const setSearch = jest.fn(() => {})
    const mockSearch = jest.fn(()=>{})

    render(<Search onChange={setSearch} searchFreelancer={mockSearch} />)

    fireEvent.change(screen.getByTestId('input'), { target: { value: 'Liz' } })
    fireEvent.click(screen.getByText('Search'))

    expect(mockSearch).toBeCalled()
    expect(mockSearch).toBeCalledTimes(1)
    
  });

  

});