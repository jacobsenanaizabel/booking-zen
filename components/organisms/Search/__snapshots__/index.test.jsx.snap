// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search tests should match snapshot with default Search 1`] = `
<DocumentFragment>
  <div
    class="styles__Container-sc-v1l7xw-0 AYMgU"
  >
    <input
      class="styles__InputStyled-sc-1i43h77-0 fULZLo"
      data-testid="input"
      name="search"
      placeholder="Find an expert by specialty or name"
      type="text"
      value=""
    />
    <div
      class="styles__Container-sc-os8su5-0"
    >
      <button
        class="styles__ButtonStyled-sc-os8su5-1 faeSvG"
        role="button"
      >
        Search
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Search tests should match snapshot with default Search with searchFreelance 1`] = `
<DocumentFragment>
  <div
    class="styles__Container-sc-v1l7xw-0 AYMgU"
  >
    <input
      class="styles__InputStyled-sc-1i43h77-0 fULZLo"
      data-testid="input"
      name="search"
      placeholder="Find an expert by specialty or name"
      type="text"
      value=""
    />
    <div
      class="styles__Container-sc-os8su5-0"
    >
      <button
        class="styles__ButtonStyled-sc-os8su5-1 faeSvG"
        role="button"
      >
        Search
      </button>
    </div>
  </div>
</DocumentFragment>
`;
